src/dvc_studio_client/post_live_metrics.py

```diff
@@ -22,26 +20,29 @@
 logger.setLevel(getenv(DVC_STUDIO_CLIENT_LOGLEVEL, "INFO").upper())
 
 
-def _get_remote_url(git_repo):
-    return git_repo.git.ls_remote("--get-url")
+def _get_remote_url() -> str:
+    from dulwich.porcelain import get_remote_repo
+    from dulwich.repo import Repo
+
+    with Repo.discover() as repo:
+        _remote, url = get_remote_repo(repo)
+        return url
 
 
 @lru_cache(maxsize=1)
 def get_studio_repo_url() -> Optional[str]:
-    studio_url = None
+    from dulwich.errors import NotGitRepository
+
     try:
-        git_repo = Repo()
-        studio_url = _get_remote_url(git_repo)
-    except GitError:
-        logger.debug("Tried to find remote url for the active branch but failed.\n")
-    finally:
-        if not studio_url:
-            logger.warning(
-                "Couldn't find a valid Studio Repo URL.\n"
-                "You can try manually setting the environment variable "
-                f"`{STUDIO_REPO_URL}`."
-            )
-        return studio_url  # noqa: B012  # pylint:disable=lost-exception
+        return _get_remote_url()
+    except (NotGitRepository, IndexError):
+        # IndexError happens when the head is detached
```

Comment:
Git does seem capable of fallback to appropriate remote when the head is detached.
So there is a difference in behaviour in this case.